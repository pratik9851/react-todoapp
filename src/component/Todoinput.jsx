import React from 'react'
import style from "./Todolist.module.css";

export default function Todoinput({add,state,handelChange}) {
    return (
        <div>
              <input
          className={style.inp}
          value={state}
          type="text"
          onChange={handelChange}
          // onKeyPress={this.handelenter.bind(this)}
          placeholder="Enter task"
        />
        <button onClick={()=>add(state)}>+</button>
            
        </div>
    )
}
