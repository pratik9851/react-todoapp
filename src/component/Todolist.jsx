import React from 'react'
import style from './todo1.module.css'

export default function Todolist({mytodolist,done,remove}) {
    return (
        <div>
            <div>
          {mytodolist.map((e) => (
            <div
              className={`${e.status === false ? style.list1 : style.list2}`}
              key={e.id}
            >
              {e.title}
              <div>
                <button onClick={()=>done(e.id)}>
                  {e.status ? "undone" : " done"}
                </button>
                <button onClick={()=>remove( e.id)}>
                  remove
                </button>
              </div>
            </div>
          ))}
        </div>
            
        </div>
    )
}


