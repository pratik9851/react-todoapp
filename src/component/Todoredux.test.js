import React from "react";
import ReactDom from "react-dom";

import Todolist from "./Todolist"
import renderer from "react-test-renderer"
import Todoinput from "./Todoinput"


it("list component renders without crashing",()=>{
    const div=document.createElement('div')

    ReactDom.render(<Todolist mytodolist={[{title:"task",status:false,id:Math.random()}]}></Todolist>,div)
})

it("list component matches snapshot",()=>{
   const tree= renderer.create(<Todolist mytodolist={[{title:"task",status:false,id:Math.random()}]} ></Todolist>,).toJSON()
   expect(tree).toMatchSnapshot()
})

it("input component renders without crashing",()=>{
    const div=document.createElement('div')

    ReactDom.render(<Todoinput></Todoinput>,div)
})

it("input component matches snapshot",()=>{
    const tree= renderer.create(<Todoinput></Todoinput>,).toJSON()
    expect(tree).toMatchSnapshot()
 })