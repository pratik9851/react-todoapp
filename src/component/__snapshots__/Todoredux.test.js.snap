// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component matches snapshot 1`] = `
<div>
  <input
    className="inp"
    placeholder="Enter task"
    type="text"
  />
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`list component matches snapshot 1`] = `
<div>
  <div>
    <div
      className="list1"
    >
      task
      <div>
        <button
          onClick={[Function]}
        >
           done
        </button>
        <button
          onClick={[Function]}
        >
          remove
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div
      className="list1"
    >
      task
      <div>
        <button
          onClick={[Function]}
        >
           done
        </button>
        <button
          onClick={[Function]}
        >
          remove
        </button>
      </div>
    </div>
  </div>
</div>
`;
